editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-top-level-elements.ts

```diff
@@ -22,26 +22,28 @@ const EmptyResult: GetParseSuccessOrTransientResult = {
 export function getParseSuccessOrTransientForFilePath(
   filePath: string,
   projectContents: ProjectContentTreeRoot,
-  openStoryboardFileNameKILLME: string | null,
-  transientFileState: TransientFileState | null,
+  transientFilesState: TransientFilesState | null,
 ): GetParseSuccessOrTransientResult {
   const projectFile = getContentsTreeFileFromString(projectContents, filePath)
   if (isTextFile(projectFile) && isParseSuccess(projectFile.fileContents.parsed)) {
     const parseSuccess = projectFile.fileContents.parsed
-    if (openStoryboardFileNameKILLME === filePath && transientFileState != null) {
+    const transientFileState: TransientFileState | null =
+      transientFilesState == null ? null : transientFilesState[filePath] ?? null
```

Comment:
I think this reads better:

```suggestion
      transientFilesState?.[filePath] ?? null
```

Reply:
I find that operator really weird.

Reply:
it _is_ a bit weird at first, but once I got used to it, it reads so much better than the conditional operator!

Reply:
You say that "at first", the sentence doesn't really need it. :)